### PR TITLE
services/horizon: Check more details when we test claiming a claimable balance.

### PR DESCRIPTION
**Pull request by Shaptic**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What
This is a simple addition to existing tests. 
It adds checks to the "claiming a claimable balance" tests to ensure that:
  - a random account for whom a CB is not destined cannot claim the CB
  - the actual balance of an account updates after claiming a CB

### Why
The former is an important "unhappy path" check for security, and the latter is a "fuller" integration test.

### CODE REVIEW

File: services/horizon/internal/integration/protocol14_test.go [hunk 1]
```diff
@@ -139,20 +139,53 @@ func runClaimingCBsTest(t *testing.T, assetType txnbuild.AssetType, predicate *x
...
+	// On success, we want the actual account to have a higher balance, too!
+	request := sdk.AccountRequest{AccountID: recipient.Address()}
+	details, err := client.AccountDetail(request)
+	assert.NoError(t, err)
+
+	// Ensure that the user's new balance includes the claimed balance, sans the
+	// transaction fee if applicable (this'll need to be updated if we change fees).
+	foundBalance := false
+	t.Logf("  balances: %+v", details.Balances)
+	for _, balance := range details.Balances {
```
Comment (abuiles) [inline services/horizon/internal/integration/protocol14_test.go:171]: While this works, it seems to me like this helpers are trying to abstract way too much, making it a bit hard to follow. Case in point, the checks below now are fixed to some given amount depending if you are sending a native asset or not  - removing the "configurability" of the helpers.

**Review by abuiles — APPROVED** [review]
ALGTM 👍

File: services/horizon/internal/integration/protocol14_test.go (hunk 1 above)
Comment (Shaptic) [reply services/horizon/internal/integration/protocol14_test.go:171]: I completely agree, it's painful to work around these, yet also painful to repeat 80% of the code for both cases. My larger PR (pending) should resolve this and make the helpers more like LEGOs bricks than a cinder block.
